### DISCUSSION TITLE
VSG particles

### DISCUSSION THREAD

**Discussion by joseluispascual** [vsg-dev/VulkanSceneGraph#1295] [Q&A]
Hi community! I've been researching about particle systems because I need them for my project. I've seen that there's no implementation for particles in VSG yet, but has anyone tried to integrate Vulkan particles examples in VulkanSceneGraph? The examples are based on GLSL shaders, so I think they could be integrated in a VSG application. Has anyone tried that? Thanks!

**Comment by robertosfield** [comment]
I haven't work on particular systems since writing precipitation and fire effects in the OSG.  My thought for the VSG has been for particular system to be done as much as possible on the GPU, and particular effects are so domain specific it'd be something provided by application developers/modellers.  

Potentially there might be art paths routes that we'd want to help support through additions to vsgXchange or another integrating 3rd party effects libraries like with how vsgImGui integrates ImGui.

I have lots of my plate with core VSG tasks so will need to defer to others in the community to pick up and run with particle effects.

**Comment by xarray** [comment]
FYI, Effekseer might be a good choice to be integrated: https://github.com/effekseer/Effekseer
It is designed to run mainly on GPU, and includes an editor for artist and integration examples for different engines.
